controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -651,8 +681,16 @@ func (r *Reconciler) checkIsComplete(ctx *pkgctx.VirtualMachinePublishRequestCon
 	return true
 }
 
-// getPublishRequestTask gets task with description id com.vmware.ovfs.LibraryItem.capture and specified actid in taskManager.
 func (r *Reconciler) getPublishRequestTask(ctx *pkgctx.VirtualMachinePublishRequestContext) (*vimtypes.TaskInfo, error) {
+	if ctx.ContentLibraryType == imgregv1.LibraryTypeInventory {
+		return r.getCloneVMTask(ctx)
+	}
+
+	return r.getCreateOVFTask(ctx)
+}
+
+// getCreateOVFTask gets task with description id com.vmware.ovfs.LibraryItem.capture and specified actid in taskManager.
+func (r *Reconciler) getCreateOVFTask(ctx *pkgctx.VirtualMachinePublishRequestContext) (*vimtypes.TaskInfo, error) {
 	actID := getPublishRequestActID(ctx.VMPublishRequest)
 	logger := ctx.Logger.WithValues("activationID", actID)
 
```

Comment:
why do we have to separate getPublishRequestTask into getCreateOVFTask() and getCreateOVFTask()? It looks like GetTasksByActID could handle both types of tasks already.